Sources/SecureXPC/Server/SecureMessageAcceptor.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  SecureMessageAcceptor.swift
+//  SecureXPC
+//
+//  Created by Josh Kaplan on 2022-01-06
+//
+
+import Foundation
+
+/// Accepts messages which meet the provided code signing requirements.
+///
+/// Uses undocumented functionality prior to macOS 11.
+internal struct SecureMessageAcceptor {
+    /// At least one of these code signing requirements must be met in order for the message to be accepted
+    internal let requirements: [SecRequirement]
+    
+    func acceptMessage(connection: xpc_connection_t, message: xpc_object_t) -> Bool {
+        // Get the code representing the client
+        var code: SecCode?
+        if #available(macOS 11, *) { // publicly documented, but only available since macOS 11
+            SecCodeCreateWithXPCMessage(message, SecCSFlags(), &code)
+        } else { // private undocumented function: xpc_connection_get_audit_token, available on prior versions of macOS
+            if var auditToken = xpc_connection_get_audit_token(connection) {
+                let tokenData = NSData(bytes: &auditToken, length: MemoryLayout.size(ofValue: auditToken))
+                let attributes = [kSecGuestAttributeAudit : tokenData] as NSDictionary
+                SecCodeCopyGuestWithAttributes(nil, attributes, SecCSFlags(), &code)
+            }
+        }
+
+        // Accept message if code is valid and meets any of the client requirements
+        var accept = false
+        if let code = code {
+            for requirement in self.requirements {
+                if SecCodeCheckValidity(code, SecCSFlags(), requirement) == errSecSuccess {
+                    accept = true
+                }
+            }
+        }
+        
+        return accept
```

Comment:
This `for` loop is approximating `contains(where:)` (but without the short-circuiting behaviour)

```suggestion
        guard let code = code else { return false }
        
        return self.requirements.contains(where: { SecCodeCheckValidity(code, SecCSFlags(), $0) == errSecSuccess }
```